scene/animation/animation_player.cpp

```diff
@@ -1205,9 +1205,13 @@ void AnimationPlayer::play(const StringName &p_name, float p_custom_blend, float
 	_stop_playing_caches();
 
 	c.current.from = &animation_set[name];
-	c.current.pos = p_from_end ? c.current.from->animation->get_length() : 0;
+
+	if (p_from_end) {
+		c.current.pos = c.current.from->animation->get_length();
```

Comment:
Thanks, the style is good now :)

Now regarding the logic, I wonder if this is correct though. If you use `play()` to play a different animation than the current one, won't that imply that it will stay at the position where the previous animation was? If so that would likely not be the expected outcome, it should start the new animation at 0.

Reply:
I see. What about doing this:

Line 1153
```suggestion
        if (String(name) == "")
	        name = playback.assigned;
	else
                playback.current.pos = 0;
```

This way the logic for passing or not the name stays on top of the method and is easily spotted.